protocol/x/listing/keeper/msg_create_market_permissionless.go

```diff
@@ -0,0 +1,40 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	"github.com/dydxprotocol/v4-chain/protocol/x/listing/types"
+)
+
+func (k msgServer) CreateMarketPermissionless(
+	goCtx context.Context,
+	msg *types.MsgCreateMarketPermissionless,
+) (*types.MsgCreateMarketPermissionlessResponse, error) {
+	ctx := lib.UnwrapSDKContext(goCtx, types.ModuleName)
+
+	// Check if the number of listed markets is above the hard cap
+	numPerpetuals := len(k.PerpetualsKeeper.GetAllPerpetuals(ctx))
```

Comment:
Why check num perpetuals instead of num markets?

Reply:
No reason. Shouldn't really make a difference